common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -913,6 +1028,57 @@ public UTF8String subStringIndex(UTF8String delim, int count) {
     }
   }
 
+  public UTF8String subStringIndex(UTF8String delim, int count, int collationId) {
+    if (CollationFactory.fetchCollation(collationId).supportsBinaryEquality) {
```

Comment:
aren't we already checking for this when we do codegen? This way we're losing the benefit of not having to worry about branching for the codegen path

Reply:
at least we can directly call `collatedSubStringIndex` in codegen

Reply:
Sure, for this function we can do that.